keccak-hash/src/lib.rs

```diff
@@ -45,13 +45,63 @@ pub fn keccak256(data: &mut [u8]) {
 	keccak256.finalize(data);
 }
 
+/// Computes in-place keccak256 hash of `data[range]`.
+///
+/// The `range` argument specifies a subslice of `data` in bytes to be hashed.
+/// The resulting hash will be written back to `data`.
+/// # Panics
+///
+/// If `range` is out of bounds.
+///
+/// # Example
+///
+/// ```
+/// let mut data = [1u8; 32];
+/// keccak_hash::keccak256_range(&mut data, 0..8);
+/// let expected = [
+///     0x54, 0x84, 0x4f, 0x69, 0xb4, 0xda, 0x4b, 0xb4, 0xa9, 0x9f, 0x24, 0x59, 0xb5, 0x11, 0xd4, 0x42,
+///     0xcc, 0x5b, 0xd2, 0xfd, 0xf4, 0xc3, 0x54, 0xd2, 0x07, 0xbb, 0x13, 0x08, 0x94, 0x43, 0xaf, 0x68,
+/// ];
+/// assert_eq!(&data, &expected);
+/// ```
+pub fn keccak256_range(data: &mut [u8], range: core::ops::Range<usize>) {
+	let mut keccak256 = Keccak::v256();
+	keccak256.update(&data[range]);
+	keccak256.finalize(data);
+}
+
 /// Computes in-place keccak512 hash of `data`.
 pub fn keccak512(data: &mut [u8]) {
 	let mut keccak512 = Keccak::v512();
 	keccak512.update(data.as_ref());
 	keccak512.finalize(data);
 }
 
+/// Computes in-place keccak512 hash of `data[range]`.
+///
+/// The `range` argument specifies a subslice of `data` in bytes to be hashed.
+/// The resulting hash will be written back to `data`.
+/// # Panics
+///
+/// If `range` is out of bounds.
+///
+/// # Example
+///
+/// ```
+/// let mut data = [1u8; 64];
+/// keccak_hash::keccak512_range(&mut data, 0..8);
+/// let expected = [
+///     0x90, 0x45, 0xc5, 0x9e, 0xd3, 0x0e, 0x1f, 0x42, 0xac, 0x35, 0xcc, 0xc9, 0x55, 0x7c, 0x77, 0x17,
+///     0xc8, 0x89, 0x3a, 0x77, 0x6c, 0xea, 0x2e, 0xf3, 0x88, 0xea, 0xe5, 0xc0, 0xea, 0x40, 0x26, 0x64,
+/// ];
+/// assert_eq!(&data[..32], &expected);
```

Comment:
note, that I'm checking here only the first 32 bytes of data, because `Debug` is not implemented for arrays of larger size :see_no_evil: